source/beamline/BioXAS/BioXASCarbonFilterFarm.cpp

```diff
@@ -0,0 +1,395 @@
+#include "BioXASCarbonFilterFarm.h"
+#include "util/AMErrorMonitor.h"
+
+BioXASCarbonFilterFarmControl::BioXASCarbonFilterFarmControl(QObject *parent) :
+	AMCompositeControl("BioXAS Carbon Filter Farm", "mm", parent)
+{
+	// Initialize local variables.
+
+	value_ = Filter::None;
+	setpoint_ = Filter::None;
+
+	upstreamActuator_ = 0;
+	downstreamActuator_ = 0;
+
+	cancelledMapper_ = new QSignalMapper(this);
+	connect( cancelledMapper_, SIGNAL(mapped(QObject*)), this, SLOT(onMoveCancelled(QObject*)) );
+
+	failedMapper_ = new QSignalMapper(this);
+	connect( failedMapper_, SIGNAL(mapped(QObject*)), this, SLOT(onMoveFailed(QObject*)) );
+
+	succeededMapper_ = new QSignalMapper(this);
+	connect( succeededMapper_, SIGNAL(mapped(QObject*)), this, SLOT(onMoveSucceeded(QObject*)) );
+
+	// Initialize inherited variables.
+	// Subclasses should initialize their own enum states and enum move states, as these will vary between beamlines.
+
+	setAllowsMovesWhileMoving(false);
+	setContextKnownDescription("Filter Farm");
+
+	// Emit enumChanged signals when connected, value changes, and setpoint changes. This is to make sure the control is viewed as an enum.
+
+	connect( this, SIGNAL(connected(bool)), this, SIGNAL(enumChanged()) );
+	connect( this, SIGNAL(valueChanged(double)), this, SIGNAL(enumChanged()) );
+	connect( this, SIGNAL(setpointChanged(double)), this, SIGNAL(enumChanged()) );
+
+	// Current settings.
+
+	updateFilter();
+}
+
+BioXASCarbonFilterFarmControl::~BioXASCarbonFilterFarmControl()
+{
+
+}
+
+bool BioXASCarbonFilterFarmControl::canStop() const
+{
+	bool result = false;
```

Comment:
Unless you think more logic is required later, you could replace this function with a ```return isConnected();``` statement.